lib/logstash/inputs/kafka.rb

```diff
@@ -266,8 +268,9 @@ def thread_runner(logstash_queue, consumer)
             end
           end
           # Manual offset commit
-          if @enable_auto_commit == "false"
+          if has_to_commit?(last_commit_time)
```

Comment:
by not committing anymore on all `poll` operations we now have two issues that must be addressed:
1. if you receive and process some events before `has_to_commit?`returns true and then no other events arrive, we'll never commit the offset because we have a guard at the start of the loop to skip if no records are returned from poll.
2. If events are processed but logstash is asked to terminate gracefully we don't commit the offset since the `stop` operation doesn't do it explicitly. Currently it relies on either commit per poll or auto commit.

Reply:
True, I only took in account my case in which I have a pretty stable flow.